packages/fuel-indexer-schema/src/db/tables.rs

```diff
@@ -400,21 +365,21 @@ impl Schema {
 
         let foreign_keys = get_foreign_keys(&root.schema)?;
 
-        Ok(Schema {
+        let mut schema = Schema {
             version: root.version,
             namespace: root.schema_name,
             identifier: root.schema_identifier,
-            query: root.query,
             types,
             fields,
             foreign_keys,
-        })
-    }
+        };
 
-    pub fn check_type(&self, type_name: &str) -> bool {
-        self.types.contains(type_name)
+        schema.register_queryroot_fields();
+
+        Ok(schema)
     }
 
+    /// Ensure the given type is included in this `Schema`'s types
```

Comment:
I think this comment needs to be above `check_type()` as this function returns the type for a field.